src/main/java/org/jenkinsci/plugins/workflow/steps/WaitForConditionStep.java

```diff
@@ -56,12 +71,13 @@
          */
         private final String id = UUID.randomUUID().toString();
         private static final float RECURRENCE_PERIOD_BACKOFF = 1.2f;
-        static final long MIN_RECURRENCE_PERIOD = 250; // ¼s
-        static final long MAX_RECURRENCE_PERIOD = 15000; // ¼min
-        long recurrencePeriod = MIN_RECURRENCE_PERIOD;
+        private final long initialRecurrencePeriod;
```

Comment:
If we deserialize instances created before this field was added (might happen for a long running pipeline if we update the plugin and then restart?), this will be 0, which looks like it will break the logic. Maybe make it non final and set it to MIN_RECURRENCE_PERIOD by default, or add a `readResolve` method?

Reply:
Exactly, this needs to be restored in `readResolve`, or the code reading it needs to handle 0.